ui/core/src/utils/text.ts

```diff
@@ -11,67 +11,40 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-import { DashboardResource, Variable, Datasource, EphemeralDashboardResource } from '../model';
+import { Resource } from '../model';
 
 /**
- * If the dashboard has a display name, return the dashboard display name
- * Else, only return the dashboard name
- * @param dashboard
+ * If the resource has a display name, return the variable display name
+ * Else, only return the resource metdata name
  */
-export function getDashboardDisplayName(dashboard: DashboardResource | EphemeralDashboardResource) {
-  return dashboard.spec.display?.name ?? dashboard.metadata.name;
-}
+export function getResourceDisplayName<T extends Resource>(resource: T): string {
+  // Variables
+  if (resource.spec.spec?.display?.name) {
```

Comment:
ouch

Reply:
Yup, display is handled differently for variables 😞